<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Proteomics Data Import • proDA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Proteomics Data Import">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">proDA</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/data-import.html">Proteomics Data Import</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/const-ae/proDA">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Proteomics Data Import</h1>
                        <h4 class="author">Constantin Ahlmann-Eltze</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/const-ae/proDA/blob/master/vignettes/data-import.Rmd"><code>vignettes/data-import.Rmd</code></a></small>
      <div class="hidden name"><code>data-import.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>After running your samples on a mass spectrometer, you want to find out if there are interesting patterns in the data. But the first challenge is how do you get the data from the files that your mass spectrometer produced into R?</p>
<p>In the following, I will describe several ways of importing data from MaxQuant. The general approaches will also be applicable to data from other tools, you will just have to adapt the column names.</p>
<div id="maxquant-file-overview" class="section level2">
<h2 class="hasAnchor">
<a href="#maxquant-file-overview" class="anchor"></a>MaxQuant File Overview</h2>
<p>MaxQuant is a popular tool for identifying, integrating, and combining MS peaks to derive peptide and protein intensities. MaxQuant produces serveral output files including <strong>proteinGroups.txt</strong>. It is usually a tab separated table with a lot of different columns, which can make it difficult to not get overwhelmed with information.</p>
<p>The most important columns that every proteinGroups.txt file contains are</p>
<ul>
<li><p><em>Protein IDs</em>: a semicolon delimited text listing all protein identifiers that match an identified set of peptides. Most of the time this is just a single protein, but sometimes proteins are so similar to each other because of gene duplication that it was not possible to differentiate them.</p></li>
<li><p><em>Majority protein IDs</em>: a semicolon delimited text that lists all proteins from the <em>Protein IDs</em> column which had more than half of their peptides identified.</p></li>
<li><p><em>Identification type [SAMPLENAME]</em>: For each sample there is one column that explains how the peptide peaks where identified. Either they were directly sequenced by the MS2 (“By MS/MS”) or by matching the m/z peak and elution timing across samples (“By matching”).</p></li>
<li><p><em>Intensity [SAMPLENAME]</em>: The combined intensity of the peptides of the protein. Missing or non-identified proteins are simply stored as <code>0</code>. In a label-free experiment, this is also often called <em>LFQ Intensity [SAMPLENAME]</em>.</p></li>
<li><p><em>iBAQ [SAMPLENAME]</em>: iBAQ is short for intensity-based absolute quantification. It is an attempt to make intensity values comparable across proteins. Usually the intensity values are only relative, which means that they are only comparable within one peptide. This is because differences in ionization and detection efficiency. For this is reason it is usually better to just compare the <em>Intensity</em> for finding differentially detected proteins.</p></li>
<li><p><em>Only identified by site</em>: Contains a “+” if the protein was only identified by a modification site. <br><em>Reverse</em>: Contains a “+” if the protein matches the reversed part of the decoy database. <br><em>Contaminant</em>: Contains a “+” if the protein is a commonly occurring contaminant. <br> These three columns are commonly used to filter out false hits.</p></li>
</ul>
<p>The full information what each column means is provided in the <em>tables.pdf</em>.</p>
</div>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<p>Our goal is to turn this complicated matrix into a useable matrix or a <code>SummarizedExperiment</code> object. There are several ways to achieve this, which I will explain in greater detail:</p>
<ol style="list-style-type: decimal">
<li>Use the base R functions (<code><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.delim()</a></code> and <code>[&lt;-</code>) to read in the data</li>
<li>Use the <code>tidyverse</code> to load the file and turn it into a useable object</li>
<li>Use the <a href="https://bioconductor.org/packages/DEP/"><code>DEP</code></a> package and the <code><a href="https://www.rdocumentation.org/packages/DEP/topics/import_MaxQuant">import_MaxQuant()</a></code> function</li>
</ol>
<p>This package comes with an example file that contains the LFQ data from a BioID experiment in <em>Drosophila melanogaster</em>. 11 different Palmitoyltransferases (short DHHC) were tagged with a promiscuous biotin ligase and all biotinylated proteins were enriched and identified using label-free mass spectrometry. The conditions are named after the tagged DHHC and the negetive control condition is called S2R for the cell line. Each condition was measured in triplicates, which means that there are a total of 36 conditions. To make the file smaller, I provide a reduced data set which only contains the first 122 rows of the data.</p>
</div>
<div id="base-r" class="section level1">
<h1 class="hasAnchor">
<a href="#base-r" class="anchor"></a>Base R</h1>
<p>The example file is located in</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, </a>
<a class="sourceLine" id="cb1-2" data-line-number="2">            <span class="dt">package =</span> <span class="st">"proDA"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; [1] "/home/constantin/R/x86_64-pc-linux-gnu-library/3.6/proDA/extdata/proteinGroups.txt"</span></a></code></pre></div>
<p>In this specific file, all spaces have been replaced with dots. This is an example how each output file from MaxQuant slightly differs. This can make it difficult to write a generic import function. Instead I will first demonstrate the most general approach which is to simply use the base R tools for loading the data and turning it into useful objects.</p>
<p>The first step is to load the full table.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">full_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.delim</a></span>(</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                <span class="dt">package =</span> <span class="st">"proDA"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5">)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(full_data))</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; [1] "Protein.IDs"                   "Majority.protein.IDs"         </span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; [3] "Peptide.counts..all."          "Peptide.counts..razor.unique."</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; [5] "Peptide.counts..unique."       "Protein.names"</span></a></code></pre></div>
<p>Next, I create a matrix of the intensity data, where each sample is a column and each protein group is a row.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># I use a regular expression (regex) to select the intensity columns</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">intensity_colnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(full_data)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grepl</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">."</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                                                <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(full_data))]</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># New data set which only contains the intensity columns</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(full_data[, intensity_colnames])</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(data) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">."</span>, <span class="st">""</span>, intensity_colnames)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">data[data <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log2</a></span>(data)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">data[<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;      CG1407.01 CG1407.02 CG1407.03 CG4676.01 CG4676.02 CG4676.03</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; [1,]        NA        NA        NA        NA        NA        NA</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt; [2,]        NA        NA        NA        NA        NA        NA</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt; [3,]        NA        NA        NA        NA  20.20120        NA</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; [4,]        NA  18.87622  18.90683        NA  18.77520        NA</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; [5,]  20.98961  20.40302  19.78941        NA  20.22682        NA</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; [6,]        NA        NA        NA        NA  19.25836        NA</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; [7,]        NA        NA        NA        NA        NA        NA</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(data) &lt;-<span class="st"> </span>full_data<span class="op">$</span>Protein.IDs</a></code></pre></div>
<p>In the next step I will create an annotation <code>data.frame</code> that contains information on the sample name, the condition and the replicate.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">annotation_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">    <span class="dt">Condition =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">d+"</span>, <span class="st">""</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">."</span>, </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                                       <span class="st">""</span>, intensity_colnames)),</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    <span class="dt">Replicate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">.[[:alnum:]]+</span><span class="ch">\\</span><span class="st">."</span>, </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                               <span class="st">""</span>, intensity_colnames)),</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(data)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(annotation_df)</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; # A tibble: 6 x 2</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;   Condition Replicate</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; 1 CG1407            1</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; 2 CG1407            2</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; 3 CG1407            3</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; 4 CG4676            1</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; 5 CG4676            2</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; 6 CG4676            3</span></a></code></pre></div>
<p>We can use this data to calculate the fit and test for differentially abundant proteins.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Not Run</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(proDA)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(data, <span class="dt">design=</span> annotation_df<span class="op">$</span>Condition, <span class="dt">col_data =</span> annotation_df)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw"><a href="../reference/test_diff.html">test_diff</a></span>(fit, <span class="dt">contrast =</span> CG1407 <span class="op">-</span><span class="st"> </span>S2R)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co"># End Not Run</span></a></code></pre></div>
<p>Optionally, we can turn the data also into a <code>SummarizedExperiment</code> or <code>MSnSet</code> object</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">se &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/RangedSummarizedExperiment-class">SummarizedExperiment</a></span>(<span class="kw">SimpleList</span>(<span class="dt">LFQ=</span>data), <span class="dt">colData=</span>annotation_df)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">rowData</a></span>(se) &lt;-<span class="st"> </span>full_data[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Only.identified.by.site"</span>, </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                             <span class="st">"Reverse"</span>, <span class="st">"Potential.contaminant"</span>)]</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">se</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; class: SummarizedExperiment </span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; dim: 122 36 </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; assays(1): LFQ</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; rownames(122): Q8IP47;Q9VJP8;Q9V435;A0A023GPQ3;Q2PDT6;Q7K540</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;   A0A023GPV6;A8JV04;Q7YU03 ... A0A0B4KGU4;Q9VHP0;M9PBB5;P09052</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;   A0A0B4KGW0;Q8IMX7-2;A0A0B4J3Z9;Q8IMX7</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; rowData names(3): Only.identified.by.site Reverse</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;   Potential.contaminant</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; colnames(36): CG1407.01 CG1407.02 ... S2R.02 S2R.03</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; colData names(2): Condition Replicate</span></a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(MSnbase)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">fData &lt;-<span class="st"> </span><span class="kw">AnnotatedDataFrame</span>(full_data[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Only.identified.by.site"</span>, </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">                                 <span class="st">"Reverse"</span>, <span class="st">"Potential.contaminant"</span>)])</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(fData) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(data)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">ms &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/MSnbase/topics/MSnSet-class">MSnSet</a></span>(data, <span class="dt">pData=</span><span class="kw">AnnotatedDataFrame</span>(annotation_df), <span class="dt">fData=</span>fData)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">ms</a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; MSnSet (storageMode: lockedEnvironment)</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; assayData: 122 features, 36 samples </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt;   element names: exprs </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; protocolData: none</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; phenoData</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt;   sampleNames: CG1407.01 CG1407.02 ... S2R.03 (36 total)</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt;   varLabels: Condition Replicate</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt;   varMetadata: labelDescription</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; featureData</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt;   featureNames: Q8IP47;Q9VJP8;Q9V435;A0A023GPQ3;Q2PDT6;Q7K540</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt;     A0A023GPV6;A8JV04;Q7YU03 ...</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt;     A0A0B4KGW0;Q8IMX7-2;A0A0B4J3Z9;Q8IMX7 (122 total)</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt;   fvarLabels: Only.identified.by.site Reverse</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt;     Potential.contaminant</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt;   fvarMetadata: labelDescription</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="co">#&gt; Annotation:  </span></a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="co">#&gt; - - - Processing information - - -</span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="co">#&gt;  MSnbase version: 2.10.0</span></a></code></pre></div>
<p>Both input types are also accepted by <code>proDA</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># Not Run</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(proDA)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(se, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>Condition <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="kw"><a href="../reference/test_diff.html">test_diff</a></span>(fit, <span class="dt">contrast =</span> ConditionCG1407 <span class="op">-</span><span class="st"> </span>ConditionS2R)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co"># End Not Run</span></a></code></pre></div>
</div>
<div id="tidyverse" class="section level1">
<h1 class="hasAnchor">
<a href="#tidyverse" class="anchor"></a>Tidyverse</h1>
<p>The tidyverse is a set of coherent R packages that provide many useful functions for common data analysis tasks. It replicates many of the functionalities already available in base R packages, but learns from its mistakes and avoids some of the suprising behaviors. For example strings are never automatically converted to factors. Another popular feature in the tidyverse is the pipe operator (<code>%&gt;%</code>) that makes it easy to chain complex transformations.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(stringr)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(readr)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tidyr)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tibble)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co"># Or short </span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co"># library(tidyverse)</span></a></code></pre></div>
<p>I first load the full data file</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co"># The read_tsv function works faster and more reliable than read.delim</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co"># But it sometimes needs help to identify the right type for each column,</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co"># because it looks only at the first 1,000 elements. </span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co"># I do that here for the Reverse column.</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5">full_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/readr/topics/read_delim">read_tsv</a></span>(</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, </a>
<a class="sourceLine" id="cb10-7" data-line-number="7">                <span class="dt">package =</span> <span class="st">"proDD"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">    <span class="dt">col_types =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/readr/topics/cols">cols</a></span>(<span class="dt">Reverse =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/readr/topics/parse_atomic">col_character</a></span>())</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb10-10" data-line-number="10"></a>
<a class="sourceLine" id="cb10-11" data-line-number="11">full_data</a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; # A tibble: 122 x 359</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt;    Protein.IDs Majority.protei… Peptide.counts.… Peptide.counts.…</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt;  1 Q8IP47;Q9V… Q8IP47;Q9VJP8;Q… 7;7;7;7;3;1      7;7;7;7;3;1     </span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="co">#&gt;  2 A0A023GPV6… A0A023GPV6;A8JV… 2;2;2            2;2;2           </span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="co">#&gt;  3 A0A023GQA5… A0A023GQA5;P241… 13;13            13;13           </span></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="co">#&gt;  4 Q1RKY1;A0A… Q1RKY1;A0A0B4LG… 3;3;3;3;3;3;3;3… 3;3;3;3;3;3;3;3…</span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="co">#&gt;  5 A0A0B4JD00… A0A0B4JD00;A8DY… 6;6;6;6;6;5;5;5… 6;6;6;6;6;5;5;5…</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="co">#&gt;  6 A0A0B4JCT8… A0A0B4JCT8;Q9V7… 2;2              2;2             </span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="co">#&gt;  7 A0A0B4LHQ4… A0A0B4LHQ4;A0A0… 3;3;3;3;3;3;3;3… 3;3;3;3;3;3;3;3…</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="co">#&gt;  8 A0A0B4JCW4… A0A0B4JCW4;Q9VH… 10;10;10         10;10;10        </span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23"><span class="co">#&gt;  9 Q9VDV4;A0A… Q9VDV4;A0A0B4JC… 4;4;3;3          4;4;3;3         </span></a>
<a class="sourceLine" id="cb10-24" data-line-number="24"><span class="co">#&gt; 10 A0A0B4JCY6… A0A0B4JCY6;Q7KS… 6;6;6;6;6;3;3;3… 6;6;6;6;6;3;3;3…</span></a>
<a class="sourceLine" id="cb10-25" data-line-number="25"><span class="co">#&gt; # … with 112 more rows, and 355 more variables:</span></a>
<a class="sourceLine" id="cb10-26" data-line-number="26"><span class="co">#&gt; #   Peptide.counts..unique. &lt;chr&gt;, Protein.names &lt;chr&gt;, Gene.names &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb10-27" data-line-number="27"><span class="co">#&gt; #   Fasta.headers &lt;chr&gt;, Number.of.proteins &lt;dbl&gt;, Peptides &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-28" data-line-number="28"><span class="co">#&gt; #   Razor...unique.peptides &lt;dbl&gt;, Unique.peptides &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-29" data-line-number="29"><span class="co">#&gt; #   Peptides.CG1407.01 &lt;dbl&gt;, Peptides.CG1407.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-30" data-line-number="30"><span class="co">#&gt; #   Peptides.CG1407.03 &lt;dbl&gt;, Peptides.CG4676.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-31" data-line-number="31"><span class="co">#&gt; #   Peptides.CG4676.02 &lt;dbl&gt;, Peptides.CG4676.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-32" data-line-number="32"><span class="co">#&gt; #   Peptides.CG51963.01 &lt;dbl&gt;, Peptides.CG51963.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-33" data-line-number="33"><span class="co">#&gt; #   Peptides.CG51963.03 &lt;dbl&gt;, Peptides.CG5620A.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-34" data-line-number="34"><span class="co">#&gt; #   Peptides.CG5620A.02 &lt;dbl&gt;, Peptides.CG5620A.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-35" data-line-number="35"><span class="co">#&gt; #   Peptides.CG5620B.01 &lt;dbl&gt;, Peptides.CG5620B.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-36" data-line-number="36"><span class="co">#&gt; #   Peptides.CG5620B.03 &lt;dbl&gt;, Peptides.CG5880.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-37" data-line-number="37"><span class="co">#&gt; #   Peptides.CG5880.02 &lt;dbl&gt;, Peptides.CG5880.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-38" data-line-number="38"><span class="co">#&gt; #   Peptides.CG6017.01 &lt;dbl&gt;, Peptides.CG6017.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-39" data-line-number="39"><span class="co">#&gt; #   Peptides.CG6017.03 &lt;dbl&gt;, Peptides.CG6618.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-40" data-line-number="40"><span class="co">#&gt; #   Peptides.CG6618.02 &lt;dbl&gt;, Peptides.CG6618.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-41" data-line-number="41"><span class="co">#&gt; #   Peptides.CG6627.01 &lt;dbl&gt;, Peptides.CG6627.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-42" data-line-number="42"><span class="co">#&gt; #   Peptides.CG6627.03 &lt;dbl&gt;, Peptides.CG8314.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-43" data-line-number="43"><span class="co">#&gt; #   Peptides.CG8314.02 &lt;dbl&gt;, Peptides.CG8314.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-44" data-line-number="44"><span class="co">#&gt; #   Peptides.GsbPI.001 &lt;dbl&gt;, Peptides.GsbPI.002 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-45" data-line-number="45"><span class="co">#&gt; #   Peptides.GsbPI.003 &lt;dbl&gt;, Peptides.S2R.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-46" data-line-number="46"><span class="co">#&gt; #   Peptides.S2R.02 &lt;dbl&gt;, Peptides.S2R.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-47" data-line-number="47"><span class="co">#&gt; #   Razor...unique.peptides.CG1407.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-48" data-line-number="48"><span class="co">#&gt; #   Razor...unique.peptides.CG1407.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-49" data-line-number="49"><span class="co">#&gt; #   Razor...unique.peptides.CG1407.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-50" data-line-number="50"><span class="co">#&gt; #   Razor...unique.peptides.CG4676.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-51" data-line-number="51"><span class="co">#&gt; #   Razor...unique.peptides.CG4676.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-52" data-line-number="52"><span class="co">#&gt; #   Razor...unique.peptides.CG4676.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-53" data-line-number="53"><span class="co">#&gt; #   Razor...unique.peptides.CG51963.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-54" data-line-number="54"><span class="co">#&gt; #   Razor...unique.peptides.CG51963.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-55" data-line-number="55"><span class="co">#&gt; #   Razor...unique.peptides.CG51963.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-56" data-line-number="56"><span class="co">#&gt; #   Razor...unique.peptides.CG5620A.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-57" data-line-number="57"><span class="co">#&gt; #   Razor...unique.peptides.CG5620A.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-58" data-line-number="58"><span class="co">#&gt; #   Razor...unique.peptides.CG5620A.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-59" data-line-number="59"><span class="co">#&gt; #   Razor...unique.peptides.CG5620B.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-60" data-line-number="60"><span class="co">#&gt; #   Razor...unique.peptides.CG5620B.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-61" data-line-number="61"><span class="co">#&gt; #   Razor...unique.peptides.CG5620B.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-62" data-line-number="62"><span class="co">#&gt; #   Razor...unique.peptides.CG5880.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-63" data-line-number="63"><span class="co">#&gt; #   Razor...unique.peptides.CG5880.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-64" data-line-number="64"><span class="co">#&gt; #   Razor...unique.peptides.CG5880.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-65" data-line-number="65"><span class="co">#&gt; #   Razor...unique.peptides.CG6017.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-66" data-line-number="66"><span class="co">#&gt; #   Razor...unique.peptides.CG6017.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-67" data-line-number="67"><span class="co">#&gt; #   Razor...unique.peptides.CG6017.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-68" data-line-number="68"><span class="co">#&gt; #   Razor...unique.peptides.CG6618.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-69" data-line-number="69"><span class="co">#&gt; #   Razor...unique.peptides.CG6618.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-70" data-line-number="70"><span class="co">#&gt; #   Razor...unique.peptides.CG6618.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-71" data-line-number="71"><span class="co">#&gt; #   Razor...unique.peptides.CG6627.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-72" data-line-number="72"><span class="co">#&gt; #   Razor...unique.peptides.CG6627.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-73" data-line-number="73"><span class="co">#&gt; #   Razor...unique.peptides.CG6627.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-74" data-line-number="74"><span class="co">#&gt; #   Razor...unique.peptides.CG8314.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-75" data-line-number="75"><span class="co">#&gt; #   Razor...unique.peptides.CG8314.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-76" data-line-number="76"><span class="co">#&gt; #   Razor...unique.peptides.CG8314.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-77" data-line-number="77"><span class="co">#&gt; #   Razor...unique.peptides.GsbPI.001 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-78" data-line-number="78"><span class="co">#&gt; #   Razor...unique.peptides.GsbPI.002 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-79" data-line-number="79"><span class="co">#&gt; #   Razor...unique.peptides.GsbPI.003 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-80" data-line-number="80"><span class="co">#&gt; #   Razor...unique.peptides.S2R.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-81" data-line-number="81"><span class="co">#&gt; #   Razor...unique.peptides.S2R.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-82" data-line-number="82"><span class="co">#&gt; #   Razor...unique.peptides.S2R.03 &lt;dbl&gt;, Unique.peptides.CG1407.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-83" data-line-number="83"><span class="co">#&gt; #   Unique.peptides.CG1407.02 &lt;dbl&gt;, Unique.peptides.CG1407.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-84" data-line-number="84"><span class="co">#&gt; #   Unique.peptides.CG4676.01 &lt;dbl&gt;, Unique.peptides.CG4676.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-85" data-line-number="85"><span class="co">#&gt; #   Unique.peptides.CG4676.03 &lt;dbl&gt;, Unique.peptides.CG51963.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-86" data-line-number="86"><span class="co">#&gt; #   Unique.peptides.CG51963.02 &lt;dbl&gt;, Unique.peptides.CG51963.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-87" data-line-number="87"><span class="co">#&gt; #   Unique.peptides.CG5620A.01 &lt;dbl&gt;, Unique.peptides.CG5620A.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-88" data-line-number="88"><span class="co">#&gt; #   Unique.peptides.CG5620A.03 &lt;dbl&gt;, Unique.peptides.CG5620B.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-89" data-line-number="89"><span class="co">#&gt; #   Unique.peptides.CG5620B.02 &lt;dbl&gt;, Unique.peptides.CG5620B.03 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-90" data-line-number="90"><span class="co">#&gt; #   Unique.peptides.CG5880.01 &lt;dbl&gt;, Unique.peptides.CG5880.02 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-91" data-line-number="91"><span class="co">#&gt; #   Unique.peptides.CG5880.03 &lt;dbl&gt;, Unique.peptides.CG6017.01 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb10-92" data-line-number="92"><span class="co">#&gt; #   Unique.peptides.CG6017.02 &lt;dbl&gt;, …</span></a></code></pre></div>
<p>Next, I create a tidy version of the data set. I pipe (<code>%&gt;%</code>) the results from each transformation to the next transformation, to first <code>select</code> the columns of interest, reshape (<code>gather</code>) the dataset from wide to long format, and lastly create new columns with <code>mutate</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">tidy_data &lt;-<span class="st"> </span>full_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/select">select</a></span>(<span class="dt">ProteinID=</span>Protein.IDs, <span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/reexports">starts_with</a></span>(<span class="st">"LFQ.intensity."</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/tidyr/topics/gather">gather</a></span>(Sample, Intensity, <span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/reexports">starts_with</a></span>(<span class="st">"LFQ.intensity."</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/mutate">mutate</a></span>(<span class="dt">Condition =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stringr/topics/str_match">str_match</a></span>(Sample, </a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                 <span class="st">"LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">.([[:alnum:]]+)</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">d+"</span>)[,<span class="dv">2</span>]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/mutate">mutate</a></span>(<span class="dt">Replicate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/stringr/topics/str_match">str_match</a></span>(Sample, </a>
<a class="sourceLine" id="cb11-7" data-line-number="7">                 <span class="st">"LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">.[[:alnum:]]+</span><span class="ch">\\</span><span class="st">.(</span><span class="ch">\\</span><span class="st">d+)"</span>)[,<span class="dv">2</span>])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/mutate">mutate</a></span>(<span class="dt">SampleName =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(Condition, <span class="st">"."</span>, Replicate))</a>
<a class="sourceLine" id="cb11-9" data-line-number="9"></a>
<a class="sourceLine" id="cb11-10" data-line-number="10">tidy_data</a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt; # A tibble: 4,392 x 6</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;    ProteinID             Sample    Intensity Condition Replicate SampleName</span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt;    &lt;chr&gt;                 &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt;  1 Q8IP47;Q9VJP8;Q9V435… LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt;  2 A0A023GPV6;A8JV04;Q7… LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co">#&gt;  3 A0A023GQA5;P24156     LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt;  4 Q1RKY1;A0A0B4LG19;A0… LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="co">#&gt;  5 A0A0B4JD00;A8DY69;I0… LFQ.inte…   2082100 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="co">#&gt;  6 A0A0B4JCT8;Q9V780     LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="co">#&gt;  7 A0A0B4LHQ4;A0A0B4JD6… LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="co">#&gt;  8 A0A0B4JCW4;Q9VHJ8;Q9… LFQ.inte…   2858600 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="co">#&gt;  9 Q9VDV4;A0A0B4JCY1;Q8… LFQ.inte…   1291400 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="co">#&gt; 10 A0A0B4JCY6;Q7KSF4;A0… LFQ.inte…         0 CG1407            1 CG1407.1  </span></a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="co">#&gt; # … with 4,382 more rows</span></a></code></pre></div>
<p>Using the tidy data, I create the annotation data frame and the data matrix.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">data &lt;-<span class="st"> </span>tidy_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/mutate">mutate</a></span>(<span class="dt">Intensity =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(Intensity <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="ot">NA</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log2</a></span>(Intensity))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/select">select</a></span>(ProteinID, SampleName, Intensity) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/tidyr/topics/spread">spread</a></span>(SampleName, Intensity) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/tibble/topics/rownames">column_to_rownames</a></span>(<span class="st">"ProteinID"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>()</a>
<a class="sourceLine" id="cb12-7" data-line-number="7"></a>
<a class="sourceLine" id="cb12-8" data-line-number="8">annotation_df &lt;-<span class="st"> </span>tidy_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/select">select</a></span>(SampleName, Condition, Replicate) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/distinct">distinct</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/dplyr/topics/arrange">arrange</a></span>(Condition, Replicate) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/tibble/topics/rownames">column_to_rownames</a></span>(<span class="st">"SampleName"</span>)</a></code></pre></div>
<p>Optionally, we can again turn this into a <code>SummarizedExperiment</code> or <code>MSnSet</code> object</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">se &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/RangedSummarizedExperiment-class">SummarizedExperiment</a></span>(<span class="kw">SimpleList</span>(<span class="dt">LFQ=</span>data), <span class="dt">colData=</span>annotation_df)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">rowData</a></span>(se) &lt;-<span class="st"> </span>full_data[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Only.identified.by.site"</span>, </a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                             <span class="st">"Reverse"</span>, <span class="st">"Potential.contaminant"</span>)]</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">se</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; class: SummarizedExperiment </span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; dim: 122 36 </span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; assays(1): LFQ</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; rownames(122): A0A023GPV6;A8JV04;Q7YU03 A0A023GQA5;P24156 ...</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt;   Q9VHX9;A0A0B4KFA6 Q9VNF8;A0A0B4K6T4;A0A0B4K5Z8</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; rowData names(3): Only.identified.by.site Reverse</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt;   Potential.contaminant</span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; colnames(36): CG1407.1 CG1407.2 ... S2R.2 S2R.3</span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; colData names(2): Condition Replicate</span></a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(MSnbase)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"></a>
<a class="sourceLine" id="cb14-3" data-line-number="3">fData &lt;-<span class="st"> </span><span class="kw">AnnotatedDataFrame</span>(full_data[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Only.identified.by.site"</span>, </a>
<a class="sourceLine" id="cb14-4" data-line-number="4">                                 <span class="st">"Reverse"</span>, <span class="st">"Potential.contaminant"</span>)])</a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/tibble/topics/rownames">rownames</a></span>(fData) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/tibble/topics/rownames">rownames</a></span>(data)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt; Warning: Setting row names on a tibble is deprecated.</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">ms &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/MSnbase/topics/MSnSet-class">MSnSet</a></span>(data, <span class="dt">pData=</span><span class="kw">AnnotatedDataFrame</span>(annotation_df), <span class="dt">fData=</span>fData)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">ms</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; MSnSet (storageMode: lockedEnvironment)</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt; assayData: 122 features, 36 samples </span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt;   element names: exprs </span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt; protocolData: none</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt; phenoData</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt;   sampleNames: CG1407.1 CG1407.2 ... S2R.3 (36 total)</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt;   varLabels: Condition Replicate</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt;   varMetadata: labelDescription</span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt; featureData</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt;   featureNames: 1 2 ... 3 (122 total)</span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"><span class="co">#&gt;   fvarLabels: Only.identified.by.site Reverse</span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#&gt;     Potential.contaminant</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#&gt;   fvarMetadata: labelDescription</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="co">#&gt; Annotation:  </span></a>
<a class="sourceLine" id="cb14-24" data-line-number="24"><span class="co">#&gt; - - - Processing information - - -</span></a>
<a class="sourceLine" id="cb14-25" data-line-number="25"><span class="co">#&gt;  MSnbase version: 2.10.0</span></a></code></pre></div>
<p>Both input types are also accepted by <code>proDA</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co"># Not Run</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(proDA)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(se, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>Condition <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw"><a href="../reference/test_diff.html">test_diff</a></span>(fit, <span class="dt">contrast =</span> ConditionCG1407 <span class="op">-</span><span class="st"> </span>ConditionS2R)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="co"># End Not Run</span></a></code></pre></div>
</div>
<div id="dep" class="section level1">
<h1 class="hasAnchor">
<a href="#dep" class="anchor"></a>DEP</h1>
<p>DEP is a BioConductor package that is designed for the analysis of mass spectrometry data. It provides helper functions to impute missing values and makes it easy to call <code>limma</code> on the result.</p>
<p>To load the data, we need to provide all the column names of the intensity values. I then call the <code><a href="https://www.rdocumentation.org/packages/DEP/topics/import_MaxQuant">import_MaxQuant()</a></code> function that directly creates a <code>SummarizedExperiment</code> object.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DEP)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#&gt; Attaching package: 'DEP'</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#&gt; The following object is masked from 'package:MSnbase':</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#&gt;     impute</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"></a>
<a class="sourceLine" id="cb16-8" data-line-number="8">full_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.delim</a></span>(</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, </a>
<a class="sourceLine" id="cb16-10" data-line-number="10">                <span class="dt">package =</span> <span class="st">"proDD"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb16-11" data-line-number="11">    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb16-13" data-line-number="13"></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"></a>
<a class="sourceLine" id="cb16-15" data-line-number="15">exp_design &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(</a>
<a class="sourceLine" id="cb16-16" data-line-number="16">    <span class="dt">label =</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"LFQ.intensity.CG1407.01"</span>, <span class="st">"LFQ.intensity.CG1407.02"</span>,  <span class="st">"LFQ.intensity.CG1407.03"</span>,  <span class="st">"LFQ.intensity.CG4676.01"</span>,  <span class="st">"LFQ.intensity.CG4676.02"</span>, <span class="st">"LFQ.intensity.CG4676.03"</span>, <span class="st">"LFQ.intensity.CG51963.01"</span>, <span class="st">"LFQ.intensity.CG51963.02"</span>, <span class="st">"LFQ.intensity.CG51963.03"</span>,<span class="st">"LFQ.intensity.CG5620A.01"</span>, <span class="st">"LFQ.intensity.CG5620A.02"</span>, <span class="st">"LFQ.intensity.CG5620A.03"</span>, <span class="st">"LFQ.intensity.CG5620B.01"</span>,<span class="st">"LFQ.intensity.CG5620B.02"</span>, <span class="st">"LFQ.intensity.CG5620B.03"</span>, <span class="st">"LFQ.intensity.CG5880.01"</span>, <span class="st">"LFQ.intensity.CG5880.02"</span>,  <span class="st">"LFQ.intensity.CG5880.03"</span>,  <span class="st">"LFQ.intensity.CG6017.01"</span>,  <span class="st">"LFQ.intensity.CG6017.02"</span>, <span class="st">"LFQ.intensity.CG6017.03"</span>, <span class="st">"LFQ.intensity.CG6618.01"</span>,  <span class="st">"LFQ.intensity.CG6618.02"</span>,  <span class="st">"LFQ.intensity.CG6618.03"</span>,  <span class="st">"LFQ.intensity.CG6627.01"</span>, <span class="st">"LFQ.intensity.CG6627.02"</span>, <span class="st">"LFQ.intensity.CG6627.03"</span>,  <span class="st">"LFQ.intensity.CG8314.01"</span>, <span class="st">"LFQ.intensity.CG8314.02"</span>,  <span class="st">"LFQ.intensity.CG8314.03"</span>, <span class="st">"LFQ.intensity.GsbPI.001"</span>, <span class="st">"LFQ.intensity.GsbPI.002"</span>,  <span class="st">"LFQ.intensity.GsbPI.003"</span>,  <span class="st">"LFQ.intensity.S2R.01"</span>, <span class="st">"LFQ.intensity.S2R.02"</span>, <span class="st">"LFQ.intensity.S2R.03"</span>),</a>
<a class="sourceLine" id="cb16-17" data-line-number="17">   <span class="dt">condition =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"CG1407"</span>, <span class="st">"CG1407"</span>, <span class="st">"CG1407"</span>, <span class="st">"CG4676"</span>, <span class="st">"CG4676"</span>, <span class="st">"CG4676"</span>, <span class="st">"CG51963"</span>, <span class="st">"CG51963"</span>, <span class="st">"CG51963"</span>, <span class="st">"CG5620A"</span>, <span class="st">"CG5620A"</span>, <span class="st">"CG5620A"</span>, <span class="st">"CG5620B"</span>, <span class="st">"CG5620B"</span>, <span class="st">"CG5620B"</span>, <span class="st">"CG5880"</span>, <span class="st">"CG5880"</span>, <span class="st">"CG5880"</span>, <span class="st">"CG6017"</span>, <span class="st">"CG6017"</span>, <span class="st">"CG6017"</span>, <span class="st">"CG6618"</span>, <span class="st">"CG6618"</span>, <span class="st">"CG6618"</span>, <span class="st">"CG6627"</span>, <span class="st">"CG6627"</span>, <span class="st">"CG6627"</span>, <span class="st">"CG8314"</span>, <span class="st">"CG8314"</span>, <span class="st">"CG8314"</span>, <span class="st">"GsbPI"</span>, <span class="st">"GsbPI"</span>, <span class="st">"GsbPI"</span>, <span class="st">"S2R"</span>, <span class="st">"S2R"</span>, <span class="st">"S2R"</span> ),</a>
<a class="sourceLine" id="cb16-18" data-line-number="18">   <span class="dt">replicate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">times=</span><span class="dv">12</span>),</a>
<a class="sourceLine" id="cb16-19" data-line-number="19">  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb16-20" data-line-number="20">)</a>
<a class="sourceLine" id="cb16-21" data-line-number="21"></a>
<a class="sourceLine" id="cb16-22" data-line-number="22">se &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DEP/topics/import_MaxQuant">import_MaxQuant</a></span>(full_data, exp_design)</a>
<a class="sourceLine" id="cb16-23" data-line-number="23"><span class="co">#&gt; Filtering based on 'Reverse', 'Potential.contaminant' column(s)</span></a>
<a class="sourceLine" id="cb16-24" data-line-number="24"><span class="co">#&gt; Making unique names</span></a>
<a class="sourceLine" id="cb16-25" data-line-number="25"><span class="co">#&gt; Obtaining SummarizedExperiment object</span></a>
<a class="sourceLine" id="cb16-26" data-line-number="26">se</a>
<a class="sourceLine" id="cb16-27" data-line-number="27"><span class="co">#&gt; class: SummarizedExperiment </span></a>
<a class="sourceLine" id="cb16-28" data-line-number="28"><span class="co">#&gt; dim: 122 36 </span></a>
<a class="sourceLine" id="cb16-29" data-line-number="29"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb16-30" data-line-number="30"><span class="co">#&gt; assays(1): ''</span></a>
<a class="sourceLine" id="cb16-31" data-line-number="31"><span class="co">#&gt; rownames(122): yuri A0A023GPV6 ... bel Miro</span></a>
<a class="sourceLine" id="cb16-32" data-line-number="32"><span class="co">#&gt; rowData names(325): Protein.IDs Majority.protein.IDs ... name ID</span></a>
<a class="sourceLine" id="cb16-33" data-line-number="33"><span class="co">#&gt; colnames(36): CG1407_1 CG1407_2 ... S2R_2 S2R_3</span></a>
<a class="sourceLine" id="cb16-34" data-line-number="34"><span class="co">#&gt; colData names(4): label ID condition replicate</span></a>
<a class="sourceLine" id="cb16-35" data-line-number="35"><span class="kw"><a href="https://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">assay</a></span>(se)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb16-36" data-line-number="36"><span class="co">#&gt;            CG1407_1 CG1407_2 CG1407_3 CG4676_1 CG4676_2</span></a>
<a class="sourceLine" id="cb16-37" data-line-number="37"><span class="co">#&gt; yuri             NA       NA       NA       NA       NA</span></a>
<a class="sourceLine" id="cb16-38" data-line-number="38"><span class="co">#&gt; A0A023GPV6       NA       NA       NA       NA       NA</span></a>
<a class="sourceLine" id="cb16-39" data-line-number="39"><span class="co">#&gt; l(2)37Cc         NA       NA       NA       NA 20.20120</span></a>
<a class="sourceLine" id="cb16-40" data-line-number="40"><span class="co">#&gt; CG10737-RC       NA 18.87622 18.90683       NA 18.77520</span></a>
<a class="sourceLine" id="cb16-41" data-line-number="41"><span class="co">#&gt; Lpin       20.98961 20.40302 19.78941       NA 20.22682</span></a></code></pre></div>
<p>Again, we can run <code>proDA</code> on the result:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co"># Not Run</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(proDA)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(se, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>condition <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co"># Here, we need to be specific, because DEP also has a test_diff method</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5">proDA<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/proDA/topics/test_diff">test_diff</a></span>(fit, <span class="dt">contrast =</span> conditionCG1407 <span class="op">-</span><span class="st"> </span>conditionS2R)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co"># End Not Run</span></a></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; R version 3.6.0 (2019-04-26)</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 18.04.2 LTS</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3</span></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so</span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb18-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb18-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb18-20" data-line-number="20"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb18-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-22" data-line-number="22"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb18-23" data-line-number="23"><span class="co">#&gt;  [1] DEP_1.6.0                   tibble_2.1.1               </span></a>
<a class="sourceLine" id="cb18-24" data-line-number="24"><span class="co">#&gt;  [3] tidyr_0.8.3                 readr_1.3.1                </span></a>
<a class="sourceLine" id="cb18-25" data-line-number="25"><span class="co">#&gt;  [5] stringr_1.4.0               dplyr_0.8.0.1              </span></a>
<a class="sourceLine" id="cb18-26" data-line-number="26"><span class="co">#&gt;  [7] MSnbase_2.10.0              ProtGenerics_1.16.0        </span></a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="co">#&gt;  [9] mzR_2.18.0                  Rcpp_1.0.1                 </span></a>
<a class="sourceLine" id="cb18-28" data-line-number="28"><span class="co">#&gt; [11] SummarizedExperiment_1.14.0 DelayedArray_0.10.0        </span></a>
<a class="sourceLine" id="cb18-29" data-line-number="29"><span class="co">#&gt; [13] BiocParallel_1.18.0         matrixStats_0.54.0         </span></a>
<a class="sourceLine" id="cb18-30" data-line-number="30"><span class="co">#&gt; [15] Biobase_2.44.0              GenomicRanges_1.36.0       </span></a>
<a class="sourceLine" id="cb18-31" data-line-number="31"><span class="co">#&gt; [17] GenomeInfoDb_1.20.0         IRanges_2.18.0             </span></a>
<a class="sourceLine" id="cb18-32" data-line-number="32"><span class="co">#&gt; [19] S4Vectors_0.22.0            BiocGenerics_0.30.0        </span></a>
<a class="sourceLine" id="cb18-33" data-line-number="33"><span class="co">#&gt; [21] BiocStyle_2.12.0           </span></a>
<a class="sourceLine" id="cb18-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb18-35" data-line-number="35"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb18-36" data-line-number="36"><span class="co">#&gt;  [1] bitops_1.0-6           fs_1.3.0               RColorBrewer_1.1-2    </span></a>
<a class="sourceLine" id="cb18-37" data-line-number="37"><span class="co">#&gt;  [4] doParallel_1.0.14      rprojroot_1.3-2        tools_3.6.0           </span></a>
<a class="sourceLine" id="cb18-38" data-line-number="38"><span class="co">#&gt;  [7] backports_1.1.4        DT_0.5                 tmvtnorm_1.4-10       </span></a>
<a class="sourceLine" id="cb18-39" data-line-number="39"><span class="co">#&gt; [10] utf8_1.1.4             R6_2.4.0               affyio_1.54.0         </span></a>
<a class="sourceLine" id="cb18-40" data-line-number="40"><span class="co">#&gt; [13] lazyeval_0.2.2         colorspace_1.4-1       GetoptLong_0.1.7      </span></a>
<a class="sourceLine" id="cb18-41" data-line-number="41"><span class="co">#&gt; [16] tidyselect_0.2.5       compiler_3.6.0         preprocessCore_1.46.0 </span></a>
<a class="sourceLine" id="cb18-42" data-line-number="42"><span class="co">#&gt; [19] cli_1.1.0              xml2_1.2.0             sandwich_2.5-1        </span></a>
<a class="sourceLine" id="cb18-43" data-line-number="43"><span class="co">#&gt; [22] desc_1.2.0             bookdown_0.9           scales_1.0.0          </span></a>
<a class="sourceLine" id="cb18-44" data-line-number="44"><span class="co">#&gt; [25] mvtnorm_1.0-10         affy_1.62.0            pkgdown_1.3.0         </span></a>
<a class="sourceLine" id="cb18-45" data-line-number="45"><span class="co">#&gt; [28] commonmark_1.7         digest_0.6.18          rmarkdown_1.12        </span></a>
<a class="sourceLine" id="cb18-46" data-line-number="46"><span class="co">#&gt; [31] XVector_0.24.0         pkgconfig_2.0.2        htmltools_0.3.6       </span></a>
<a class="sourceLine" id="cb18-47" data-line-number="47"><span class="co">#&gt; [34] limma_3.40.0           htmlwidgets_1.3        GlobalOptions_0.1.0   </span></a>
<a class="sourceLine" id="cb18-48" data-line-number="48"><span class="co">#&gt; [37] rlang_0.3.4            rstudioapi_0.10        impute_1.58.0         </span></a>
<a class="sourceLine" id="cb18-49" data-line-number="49"><span class="co">#&gt; [40] shiny_1.3.2            shape_1.4.4            zoo_1.8-5             </span></a>
<a class="sourceLine" id="cb18-50" data-line-number="50"><span class="co">#&gt; [43] mzID_1.22.0            RCurl_1.95-4.12        magrittr_1.5          </span></a>
<a class="sourceLine" id="cb18-51" data-line-number="51"><span class="co">#&gt; [46] GenomeInfoDbData_1.2.1 MALDIquant_1.19.2      Matrix_1.2-17         </span></a>
<a class="sourceLine" id="cb18-52" data-line-number="52"><span class="co">#&gt; [49] munsell_0.5.0          fansi_0.4.0            imputeLCMD_2.0        </span></a>
<a class="sourceLine" id="cb18-53" data-line-number="53"><span class="co">#&gt; [52] vsn_3.52.0             stringi_1.4.3          yaml_2.2.0            </span></a>
<a class="sourceLine" id="cb18-54" data-line-number="54"><span class="co">#&gt; [55] MASS_7.3-51.1          zlibbioc_1.30.0        plyr_1.8.4            </span></a>
<a class="sourceLine" id="cb18-55" data-line-number="55"><span class="co">#&gt; [58] grid_3.6.0             promises_1.0.1         shinydashboard_0.7.1  </span></a>
<a class="sourceLine" id="cb18-56" data-line-number="56"><span class="co">#&gt; [61] crayon_1.3.4           lattice_0.20-38        circlize_0.4.6        </span></a>
<a class="sourceLine" id="cb18-57" data-line-number="57"><span class="co">#&gt; [64] hms_0.4.2              knitr_1.22             ComplexHeatmap_2.0.0  </span></a>
<a class="sourceLine" id="cb18-58" data-line-number="58"><span class="co">#&gt; [67] pillar_1.3.1           rjson_0.2.20           codetools_0.2-16      </span></a>
<a class="sourceLine" id="cb18-59" data-line-number="59"><span class="co">#&gt; [70] XML_3.98-1.19          glue_1.3.1             evaluate_0.13         </span></a>
<a class="sourceLine" id="cb18-60" data-line-number="60"><span class="co">#&gt; [73] pcaMethods_1.76.0      BiocManager_1.30.4     httpuv_1.5.1          </span></a>
<a class="sourceLine" id="cb18-61" data-line-number="61"><span class="co">#&gt; [76] png_0.1-7              foreach_1.4.4          gtable_0.3.0          </span></a>
<a class="sourceLine" id="cb18-62" data-line-number="62"><span class="co">#&gt; [79] purrr_0.3.2            norm_1.0-9.5           clue_0.3-57           </span></a>
<a class="sourceLine" id="cb18-63" data-line-number="63"><span class="co">#&gt; [82] assertthat_0.2.1       ggplot2_3.1.1          xfun_0.6              </span></a>
<a class="sourceLine" id="cb18-64" data-line-number="64"><span class="co">#&gt; [85] mime_0.6               xtable_1.8-4           later_0.8.0           </span></a>
<a class="sourceLine" id="cb18-65" data-line-number="65"><span class="co">#&gt; [88] roxygen2_6.1.1         ncdf4_1.16.1           gmm_1.6-2             </span></a>
<a class="sourceLine" id="cb18-66" data-line-number="66"><span class="co">#&gt; [91] iterators_1.0.10       memoise_1.1.0          cluster_2.0.8</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#maxquant-file-overview">MaxQuant File Overview</a></li>
      </ul>
</li>
      <li><a href="#workflow">Workflow</a></li>
      <li><a href="#base-r">Base R</a></li>
      <li><a href="#tidyverse">Tidyverse</a></li>
      <li><a href="#dep">DEP</a></li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Constantin Ahlmann-Eltze.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
