<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • proDA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">proDA</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/data-import.html">Proteomics Data Import</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/const-ae/proDA">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction</h1>
                        <h4 class="author">Constantin Ahlmann-Eltze</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/const-ae/proDA/blob/master/vignettes/Introduction.Rmd"><code>vignettes/Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>Introduction.Rmd</code></div>

    </div>

    
    
<style>
    body .main-container {
        max-width: 1024px;
    }
</style>
<div id="proda" class="section level1">
<h1 class="hasAnchor">
<a href="#proda" class="anchor"></a>proDA</h1>
<!-- badges: start -->
<!-- badges: end -->
<p>The goal of <code>proDA</code> is to identify differentially abundant proteins in label-free mass spectrometry data. The main challenge of this data are the many missing values. The missing values don’t occur randomly but especially at low intensities. This means that they cannot just be ignored. Existing methods have mostly focused on replacing the missing values with some reasonable number (“imputation”) and then run classical methods. But imputation is problematic because it obscures the amount of available information. Which in turn can lead to over-confident predictions.</p>
<p><code>proDA</code> on the other hand does not impute missing values, but constructs a probabilistic dropout model. For each sample it fits a sigmoidal dropout curve. This information can then be used to infer means across samples and the associated uncertainty, without the intermediate imputation step. <code>proDA</code> supports full linear models with variance and location moderation.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the development version from <a href="https://github.com/const-ae/proDA">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># install.packages("devtools")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"const-ae/proDA"</span>)</a></code></pre></div>
<p>The pkgdown documentation for the package is available on <a href="https://const-ae.github.io/proDA/reference" class="uri">https://const-ae.github.io/proDA/reference</a></p>
<p>In the following section, I will give a very brief overview on the main functionality of the <code>proDA</code> package. New users are advised to directly go to section two, where I give a complete walkthrough and explain in detail, what steps are necessary for the analysis of label-free mass spectrometry data.</p>
</div>
<div id="quickstart" class="section level2">
<h2 class="hasAnchor">
<a href="#quickstart" class="anchor"></a>Quickstart</h2>
<p>The three steps that are necessary to analyze the data are</p>
<ol style="list-style-type: decimal">
<li>Load the data (see vignette on loading MaxQuant output files)</li>
<li>Fit the probabilistic dropout model (<code><a href="../reference/proDA.html">proDA()</a></code>)</li>
<li>Test in which proteins the coefficients of the model differ (<code><a href="../reference/test_diff.html">test_diff()</a></code>)</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Load the package</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(proDA)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># Generate some dataset with known structure</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">syn_dataset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_synthetic_data.html">generate_synthetic_data</a></span>(<span class="dt">n_proteins =</span> <span class="dv">100</span>, <span class="dt">n_conditions =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co"># The abundance matrix</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">syn_dataset<span class="op">$</span>Y[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ]</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt;           Condition_1-1 Condition_1-2 Condition_1-3 Condition_2-1 Condition_2-2 Condition_2-3</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt; protein_1            NA            NA      18.88592            NA      18.72059      20.06119</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt; protein_2      21.37123      20.53557      18.83239      20.41027      21.73266      21.16719</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; protein_3            NA      18.77742      18.98681            NA            NA      19.20291</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt; protein_4      25.44209      25.15151      25.38142      25.22754      24.95229      24.97185</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt; protein_5      23.46724      23.15808      23.21357      23.29562      23.25999      23.57925</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co"># Assignment of the samples to the two conditions</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16">syn_dataset<span class="op">$</span>groups</a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; [1] Condition_1 Condition_1 Condition_1 Condition_2 Condition_2 Condition_2</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; Levels: Condition_1 Condition_2</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co"># Fit the probabilistic dropout model</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(syn_dataset<span class="op">$</span>Y, <span class="dt">design =</span> syn_dataset<span class="op">$</span>groups)</a>
<a class="sourceLine" id="cb2-22" data-line-number="22"></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co"># Identify which proteins differ between Condition 1 and 2</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="kw"><a href="../reference/test_diff.html">test_diff</a></span>(fit, <span class="st">`</span><span class="dt">Condition_1</span><span class="st">`</span> <span class="op">-</span><span class="st"> `</span><span class="dt">Condition_2</span><span class="st">`</span>, <span class="dt">sort_by =</span> <span class="st">"pval"</span>, <span class="dt">n_max =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt; # A tibble: 5 x 10</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt;   name              pval adj_pval  diff t_statistic    se    df avg_abundance n_approx n_obs</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt;   &lt;chr&gt;            &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;         &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="co">#&gt; 1 protein_96  0.00000248 0.000248  8.62        39.4 0.219     4          22.2     4.02     4</span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29"><span class="co">#&gt; 2 protein_95  0.0000103  0.000513 -4.84       -27.6 0.175     4          21.2     6.       6</span></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#&gt; 3 protein_91  0.0000528  0.00176  -4.17       -18.3 0.228     4          19.1     4.01     4</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co">#&gt; 4 protein_98  0.000236   0.00479   4.35        12.5 0.348     4          21.6     6.00     6</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"><span class="co">#&gt; 5 protein_100 0.000239   0.00479   2.49        12.5 0.200     4          21.3     4.95     5</span></a></code></pre></div>
<p>Other helpful functions for quality control are <code><a href="../reference/median_normalization.html">median_normalization()</a></code> and <code><a href="../reference/dist_approx.html">dist_approx()</a></code>.</p>
</div>
<div id="proda-walkthrough" class="section level2">
<h2 class="hasAnchor">
<a href="#proda-walkthrough" class="anchor"></a>proDA Walkthrough</h2>
<p><code>proDA</code> is an R package that implements a powerful probabilistic dropout model to identify differentially abundant proteins. The package was specifically designed for label-free mass spectrometry data and in particular how to handle the many many missing values.</p>
<p>But all this is useless if you cannot load your data and get it into a shape that is useable. In the next section, I will explain how to load the abundance matrix and bring it into a useful form. The steps that I will go through are</p>
<ol style="list-style-type: decimal">
<li>Load the <code>proteinGroups.txt</code> MaxQuant output table</li>
<li>Extract the intensity columns and create the abundance matrix</li>
<li>Replace the zeros with <code>NA</code>s and take the <code><a href="https://www.rdocumentation.org/packages/base/topics/Log">log2()</a></code> of the data</li>
<li>Normalize the data using <code><a href="../reference/median_normalization.html">median_normalization()</a></code>
</li>
<li>Inspect sample structure with a heatmap of the distance matrix (<code><a href="../reference/dist_approx.html">dist_approx()</a></code>)</li>
<li>Fit the probabilisitc dropout model with <code><a href="../reference/proDA.html">proDA()</a></code>
</li>
<li>Identify differentially abundant proteins with <code><a href="../reference/test_diff.html">test_diff()</a></code>
</li>
</ol>
<div id="load-data" class="section level3">
<h3 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load Data</h3>
<p>I will now demonstrate how to load a MaxQuant output file. For more information about other approaches for loading the data, please take a look at the vignette on loading data.</p>
<p>MaxQuant is one of the most popular tools for handling raw MS data. It produces a number of files. The important file that contains the protein intensities is called <code>proteinGroups.txt</code>. It is a large table with detailed information about the identification and quantification process for each protein group (which I will from now on just call “protein”).</p>
<p>This package comes with an example <code>proteinGroups.txt</code> file, located in the package folder. The file contains the reduced output from an experiment studying the different DHHCs in Drosophila melanogaster.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, <span class="dt">package =</span> <span class="st">"proDA"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; [1] "/home/constantin/R/x86_64-pc-linux-gnu-library/3.6/proDA/extdata/proteinGroups.txt"</span></a></code></pre></div>
<p>In this example, I will use the base R functions to load the data, because they don’t require any additional dependencies.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Load the table into memory</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">maxquant_protein_table &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.delim</a></span>(</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">    <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata/proteinGroups.txt"</span>, <span class="dt">package =</span> <span class="st">"proDA"</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">)</a></code></pre></div>
<p>As I have mentioned, the table contains a lot of information (359 columns!!), but we are first of all interested in the columns which contain the measured intensities.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># I use a regular expression (regex) to select the intensity columns</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">intensity_colnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">."</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(maxquant_protein_table), <span class="dt">value=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(intensity_colnames)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; [1] "LFQ.intensity.CG1407.01" "LFQ.intensity.CG1407.02" "LFQ.intensity.CG1407.03"</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; [4] "LFQ.intensity.CG4676.01" "LFQ.intensity.CG4676.02" "LFQ.intensity.CG4676.03"</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co"># Create the intensity matrix</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9">abundance_matrix &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(maxquant_protein_table[, intensity_colnames])</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co"># Adapt column and row maxquant_protein_table</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(abundance_matrix) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"^LFQ</span><span class="ch">\\</span><span class="st">.intensity</span><span class="ch">\\</span><span class="st">."</span>, <span class="st">""</span>, intensity_colnames)</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(abundance_matrix) &lt;-<span class="st"> </span>maxquant_protein_table<span class="op">$</span>Protein.IDs</a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co"># Print some rows of the matrix with short names so they fit on the screen</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14">abundance_matrix[<span class="dv">46</span><span class="op">:</span><span class="dv">48</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt;                                       CG1407.01 CG1407.02 CG1407.03 CG4676.01 CG4676.02 CG4676.03</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; A0A0B4K6W1;P08970                        713400    845440         0         0   1032600         0</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; A0A0B4K6W2;A0A0B4K7S0;P55824-3;P55824   5018800   4429500   2667200         0   8780200   1395800</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; A0A0B4K6X7;A1Z8J0                             0         0         0         0         0         0</span></a></code></pre></div>
<p>After extracting the bits from the table we most care about, we will have to modify it.</p>
<p>Firstly, MaxQuant codes missing values as <code>0</code>. This is misleading, because the actual abundance probably was not zero, but just some value too small to be detected by the mass spectrometer. Accordingly, I will replace all <code>0</code> with <code>NA</code>.</p>
<p>Secondly, the raw intensity values have a linear mean-variance relation. This is undesirable, because a change of <code>x</code> units can be a large shift if the mean is small or irrelevant if the mean is large. Luckily, to make the mean and variance independent, we can just <code>log</code> the intensities. Now a change of <code>x</code> units is as significant for highly abundant proteins, as it is for low abundant ones.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">abundance_matrix[abundance_matrix <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">abundance_matrix &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log2</a></span>(abundance_matrix)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">abundance_matrix[<span class="dv">46</span><span class="op">:</span><span class="dv">48</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt;                                       CG1407.01 CG1407.02 CG1407.03 CG4676.01 CG4676.02 CG4676.03</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; A0A0B4K6W1;P08970                      19.44435  19.68934        NA        NA  19.97785        NA</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; A0A0B4K6W2;A0A0B4K7S0;P55824-3;P55824  22.25891  22.07871  21.34689        NA  23.06582  20.41266</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; A0A0B4K6X7;A1Z8J0                            NA        NA        NA        NA        NA        NA</span></a></code></pre></div>
</div>
<div id="quality-control" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality Control</h3>
<p>Quality control (QC) is essential for a successful bioinformatics analysis, because any dataset shows some unwanted variation or could even contain more serious error like for example a sample swap.</p>
<p>Often we start with normalizing the data to remove potential sample specific effects. But already this step is challenging, because the missing values cannot easily be corrected for. Thus, a first helpful plot is to look how many missing values are in each sample.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/barplot">barplot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">colSums</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(abundance_matrix)),</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">        <span class="dt">ylab =</span> <span class="st">"# missing values"</span>,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">        <span class="dt">xlab =</span> <span class="st">"Sample 1 to 36"</span>)</a></code></pre></div>
<p><img src="Introduction_files/figure-html/qc-mis_barplot-1.png" width="80%" style="display: block; margin: auto;"></p>
<p>We can see that the number of missing values differs substantially between samples (between 30% and 90%) in this dataset. If we take a look at the intensity distribution for each sample, we see that they differ substantially as well.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(abundance_matrix,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">        <span class="dt">ylab =</span> <span class="st">"Intensity Distribution"</span>,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">        <span class="dt">xlab =</span> <span class="st">"Sample 1 to 36"</span>)</a></code></pre></div>
<p><img src="Introduction_files/figure-html/qc-raw_boxplot-1.png" width="80%" style="display: block; margin: auto;"></p>
<p>Note that, the intensity distribution is shifted upwards for samples which also have a large number of missing values (for example the last one). This agrees with our idea that small values are more likely to be missing. On the other hand, this also demonstrates why normalization methods such as quantile normalization, which distort the data until all the distributions are equal, are problematic. I will apply the more “conservative” median normalization, which ignores the missing values and transforms the values so that the median difference between the sample and average across all other samples is zero.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">normalized_abundance_matrix &lt;-<span class="st"> </span><span class="kw"><a href="../reference/median_normalization.html">median_normalization</a></span>(abundance_matrix)</a></code></pre></div>
<p>An important tool to identify sample swaps and outliers in the dataset is to look at the sample distance matrix. It shows the distances of samples A to B, A to C, B to C and so on.</p>
<p>The base R <code><a href="https://www.rdocumentation.org/packages/stats/topics/dist">dist()</a></code> function can not handle input data that contains missing values, so we might be tempted to just replace the missing values with some realistic numbers and calculate the distance on the completed dataset. But choosing a good replacement value is challenging and can also be misleading because the samples with many missing values would be considered too close.</p>
<p>Instead <code>proDA</code> provides the <code><a href="../reference/dist_approx.html">dist_approx()</a></code> function that takes either a fitted model (ie. the output from <code><a href="../reference/proDA.html">proDA()</a></code>) or a simple matrix (for which it internally calls <code><a href="../reference/proDA.html">proDA()</a></code>) and estimates the expected distance without imputing the missing values. In addition, it reports the associated uncertainty with every estimate. The estimates for samples with many missing values will be uncertain, allowing the data analyst to discount them.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">da &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dist_approx.html">dist_approx</a></span>(normalized_abundance_matrix)</a></code></pre></div>
<p><code><a href="../reference/dist_approx.html">dist_approx()</a></code> returns two elements the <code>mean</code> of the estimate and the associated <code>sd</code>. In the next step I will plot the heatmap for three differnet conditions, adding the 95% confidence interval as text to each cell.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># This chunk only works if pheatmap is installed</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co"># install.packages("pheatmap")</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3">sel &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,  <span class="co"># CG1407</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">         <span class="dv">7</span><span class="op">:</span><span class="dv">9</span>,  <span class="co"># CG59163</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">         <span class="dv">22</span><span class="op">:</span><span class="dv">24</span>)<span class="co"># CG6618</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"></a>
<a class="sourceLine" id="cb11-7" data-line-number="7">plot_mat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(da<span class="op">$</span>mean)[sel, sel]</a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co"># Remove diagonal elements, so that the colorscale is not distorted</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9">plot_mat[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">9</span>) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co"># 95% conf interval is approx `sd * 1.96`</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11">uncertainty &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">" ± "</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Round">round</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(da<span class="op">$</span>sd <span class="op">*</span><span class="st"> </span><span class="fl">1.96</span>)[sel, sel], <span class="dv">1</span>)), <span class="dt">nrow=</span><span class="dv">9</span>)</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">pheatmap<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pheatmap/topics/pheatmap">pheatmap</a></span>(plot_mat, </a>
<a class="sourceLine" id="cb11-13" data-line-number="13">                   <span class="dt">cluster_rows =</span> <span class="ot">FALSE</span>, <span class="dt">cluster_cols =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb11-14" data-line-number="14">                   <span class="dt">display_numbers=</span> uncertainty,</a>
<a class="sourceLine" id="cb11-15" data-line-number="15">                   <span class="dt">number_color =</span> <span class="st">"black"</span>)</a></code></pre></div>
<p><img src="Introduction_files/figure-html/sample_dist-1.png" width="60%" style="display: block; margin: auto;"></p>
</div>
<div id="fit-the-probabilistic-dropout-model" class="section level3">
<h3 class="hasAnchor">
<a href="#fit-the-probabilistic-dropout-model" class="anchor"></a>Fit the Probabilistic Dropout Model</h3>
<p>In the next step, we will fit the actual linear probabilistic dropoout model to the normalized data. But before we start, I will create a data.frame that contains some additional information on each sample, in particular to which condition that sample belongs.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># The best way to create this data.frame depends on the column naming scheme</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">sample_info_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">name =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(normalized_abundance_matrix),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                             <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">sample_info_df<span class="op">$</span>condition &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/substr">substr</a></span>(sample_info_df<span class="op">$</span>name, <span class="dv">1</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nchar">nchar</a></span>(sample_info_df<span class="op">$</span>name)  <span class="op">-</span><span class="st"> </span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">sample_info_df<span class="op">$</span>replicate &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/substr">substr</a></span>(sample_info_df<span class="op">$</span>name, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nchar">nchar</a></span>(sample_info_df<span class="op">$</span>name)  <span class="op">-</span><span class="st"> </span><span class="dv">1</span>, <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb12-8" data-line-number="8">sample_info_df</a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; # A tibble: 36 x 3</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt;    name       condition replicate</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt;  1 CG1407.01  CG1407            1</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt;  2 CG1407.02  CG1407            2</span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt;  3 CG1407.03  CG1407            3</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt;  4 CG4676.01  CG4676            1</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt;  5 CG4676.02  CG4676            2</span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17"><span class="co">#&gt;  6 CG4676.03  CG4676            3</span></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="co">#&gt;  7 CG51963.01 CG51963           1</span></a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="co">#&gt;  8 CG51963.02 CG51963           2</span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20"><span class="co">#&gt;  9 CG51963.03 CG51963           3</span></a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="co">#&gt; 10 CG5620A.01 CG5620A           1</span></a>
<a class="sourceLine" id="cb12-22" data-line-number="22"><span class="co">#&gt; # … with 26 more rows</span></a></code></pre></div>
<p>Now we can call the <code><a href="../reference/proDA.html">proDA()</a></code> function to actually fit the model. We specify the <code>design</code> using the formula notation, referencing the <code>condition</code> column in the <code>sample_info_df</code> data.frame that we have just created. In addition, I specify that I want to use the <code>S2R</code> condition as the reference because I know that it was the negative control and this way automatically all coefficients measure how much each condition differs from the negative control.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proDA.html">proDA</a></span>(normalized_abundance_matrix, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>condition, </a>
<a class="sourceLine" id="cb13-2" data-line-number="2">             <span class="dt">col_data =</span> sample_info_df, <span class="dt">reference_level =</span> <span class="st">"S2R"</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">fit</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#&gt;  Parameters of the probabilistic dropout model</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; The dataset contains 36 samples and 122 proteins</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; 59.7% of the values are missing</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; Experimental design: y~condition</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; The model has successfully converged.</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; The inferred parameters are:</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt; location_prior_mean:     19.5</span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; location_prior_scale:    8.37</span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; location_prior_df:       3</span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; variance_prior_scale:    0.283</span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; variance_prior_df:       1.64</span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; dropout_curve_position:  19.9, 19, 20.1, 22.8, ...</span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; dropout_curve_scale:     -0.816, -0.601, -1.02, -1.3, ...</span></a></code></pre></div>
<p>The <code>proDAFit</code> object prints a number of useful information about the convergence of the model, the size of the dataset, the number of missing values, and the inferred hyper parameters.</p>
<p>To make it easy to find available methods on the <code>proDAFit</code> object, the <code>$</code>-operator is overloaded and shows a list of possible functions:</p>
<div class="figure">
<img src="../man/figures/README-screenshot_fit_functions.png" alt="Screenshot from Rstudio suggesting the available functions"><p class="caption">Screenshot from Rstudio suggesting the available functions</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co"># Equivalent to feature_parameters(fit)</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2">fit<span class="op">$</span>feature_parameters</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; # A tibble: 122 x 4</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#&gt;    n_approx     df       s2 n_obs</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#&gt;       &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt;  1     12.0  0.001 3808.        5</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt;  2     12.0  0.001 2439.        1</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt;  3     19.3  8.93     4.07     14</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt;  4     12.0  0.001  850.        6</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt;  5     17.4  7.04     0.470    17</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt;  6     12.0  0.001 2472.        1</span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt;  7     12.0  0.001 2410.        1</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt;  8     28.9 18.6      0.217    29</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt;  9     12.0  0.001 1797.        4</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt; 10     12.0  0.001 1881.        4</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt; # … with 112 more rows</span></a></code></pre></div>
<p>Internally the <code>proDAFit</code> object is implemented as a subclass of <code>SummarizedExperiment</code>. This means it can be subsetted, which is for example useful for calculating the distance of a subset of proteins and samples.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co"># This chunk only works if pheatmap is installed</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="co"># install.packages("pheatmap")</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">pheatmap<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pheatmap/topics/pheatmap">pheatmap</a></span>(<span class="kw"><a href="../reference/dist_approx.html">dist_approx</a></span>(fit[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="dt">by_sample =</span> <span class="ot">FALSE</span>)<span class="op">$</span>mean)</a></code></pre></div>
<p><img src="Introduction_files/figure-html/protein_dist-1.png" width="60%" style="display: block; margin: auto;"></p>
</div>
<div id="identify-differential-abundance" class="section level3">
<h3 class="hasAnchor">
<a href="#identify-differential-abundance" class="anchor"></a>Identify Differential Abundance</h3>
<p>Lastly, we will use a Wald test to identify in which proteins a coefficient is significantly different from zero. The <code><a href="../reference/test_diff.html">test_diff()</a></code> function takes first the fit object produced by <code><a href="../reference/proDA.html">proDA()</a></code> and a contrast argument. This can either be a string or an expression if we want to test more complex combinations. For example <code>conditionCG1407 - (conditionCG6017 + conditionCG5880) / 2</code> would test for the difference between CG1407 and the average of CG6017 and CG5880.</p>
<p>Alternatively <code><a href="../reference/test_diff.html">test_diff()</a></code> also supports likelihood ratio F-tests. In that case instead of the <code>contrast</code> argument specify the <code>reduced_model</code> argument.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co"># Test which proteins differ between condition CG1407 and S2R</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co"># S2R is the default contrast, because it was specified as the `reference_level`</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3">test_res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/test_diff.html">test_diff</a></span>(fit, <span class="st">"conditionCG1407"</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">test_res</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt; # A tibble: 122 x 10</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#&gt;    name                   pval adj_pval    diff t_statistic    se    df avg_abundance n_approx n_obs</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;                 &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;         &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co">#&gt;  1 Q8IP47;Q9VJP8;Q9V43… 0.904     0.964 -0.132      -0.122  1.08     24          18.9     12.0     5</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="co">#&gt;  2 A0A023GPV6;A8JV04;Q… 0.923     0.964 -0.0993     -0.0979 1.01     24          18.4     12.0     1</span></a>
<a class="sourceLine" id="cb16-10" data-line-number="10"><span class="co">#&gt;  3 A0A023GQA5;P24156    0.0356    0.265 -2.92       -2.23   1.31     24          19.3     19.3    14</span></a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="co">#&gt;  4 Q1RKY1;A0A0B4LG19;A… 0.667     0.964  0.632       0.435  1.45     24          18.7     12.0     6</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12"><span class="co">#&gt;  5 A0A0B4JD00;A8DY69;I… 0.919     0.964  0.0689      0.103  0.670    24          20.0     17.4    17</span></a>
<a class="sourceLine" id="cb16-13" data-line-number="13"><span class="co">#&gt;  6 A0A0B4JCT8;Q9V780    0.923     0.964 -0.0995     -0.0981 1.01     24          18.5     12.0     1</span></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"><span class="co">#&gt;  7 A0A0B4LHQ4;A0A0B4JD… 0.923     0.964 -0.0991     -0.0977 1.01     24          18.4     12.0     1</span></a>
<a class="sourceLine" id="cb16-15" data-line-number="15"><span class="co">#&gt;  8 A0A0B4JCW4;Q9VHJ8;Q… 0.643     0.964 -0.197      -0.469  0.419    24          21.9     28.9    29</span></a>
<a class="sourceLine" id="cb16-16" data-line-number="16"><span class="co">#&gt;  9 Q9VDV4;A0A0B4JCY1;Q… 0.295     0.861  1.95        1.07   1.82     24          18.7     12.0     4</span></a>
<a class="sourceLine" id="cb16-17" data-line-number="17"><span class="co">#&gt; 10 A0A0B4JCY6;Q7KSF4;A… 0.598     0.964 -0.783      -0.535  1.46     24          19.0     12.0     4</span></a>
<a class="sourceLine" id="cb16-18" data-line-number="18"><span class="co">#&gt; # … with 112 more rows</span></a></code></pre></div>
<p>This walkthrough ends with the identification which proteins are differentially abundant. But for a real dataset, now the actual analysis only just begins. A list of significant proteins is hardly ever a publishable result, one often needs to make sense what the relevant underlying biological mechanisms are. But for this problem other tools are necessary, which depend on the precise question associated with the biological problem at hand.</p>
</div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#&gt; R version 3.6.0 (2019-04-26)</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 18.04.2 LTS</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3</span></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8       </span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="co">#&gt;  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt;  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                  LC_ADDRESS=C              </span></a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; [10] LC_TELEPHONE=C             LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="co">#&gt; [1] proDA_0.99.0     BiocStyle_2.12.0</span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="co">#&gt;  [1] SummarizedExperiment_1.14.0 xfun_0.6                    lattice_0.20-38            </span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="co">#&gt;  [4] colorspace_1.4-1            htmltools_0.3.6             stats4_3.6.0               </span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="co">#&gt;  [7] yaml_2.2.0                  utf8_1.1.4                  rlang_0.3.4                </span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="co">#&gt; [10] pkgdown_1.3.0               pillar_1.3.1                BiocParallel_1.18.0        </span></a>
<a class="sourceLine" id="cb17-27" data-line-number="27"><span class="co">#&gt; [13] BiocGenerics_0.30.0         RColorBrewer_1.1-2          matrixStats_0.54.0         </span></a>
<a class="sourceLine" id="cb17-28" data-line-number="28"><span class="co">#&gt; [16] GenomeInfoDbData_1.2.1      stringr_1.4.0               zlibbioc_1.30.0            </span></a>
<a class="sourceLine" id="cb17-29" data-line-number="29"><span class="co">#&gt; [19] munsell_0.5.0               gtable_0.3.0                commonmark_1.7             </span></a>
<a class="sourceLine" id="cb17-30" data-line-number="30"><span class="co">#&gt; [22] memoise_1.1.0               evaluate_0.13               Biobase_2.44.0             </span></a>
<a class="sourceLine" id="cb17-31" data-line-number="31"><span class="co">#&gt; [25] knitr_1.22                  IRanges_2.18.0              GenomeInfoDb_1.20.0        </span></a>
<a class="sourceLine" id="cb17-32" data-line-number="32"><span class="co">#&gt; [28] parallel_3.6.0              fansi_0.4.0                 Rcpp_1.0.1                 </span></a>
<a class="sourceLine" id="cb17-33" data-line-number="33"><span class="co">#&gt; [31] scales_1.0.0                backports_1.1.4             BiocManager_1.30.4         </span></a>
<a class="sourceLine" id="cb17-34" data-line-number="34"><span class="co">#&gt; [34] DelayedArray_0.10.0         S4Vectors_0.22.0            desc_1.2.0                 </span></a>
<a class="sourceLine" id="cb17-35" data-line-number="35"><span class="co">#&gt; [37] XVector_0.24.0              fs_1.3.0                    digest_0.6.18              </span></a>
<a class="sourceLine" id="cb17-36" data-line-number="36"><span class="co">#&gt; [40] stringi_1.4.3               bookdown_0.9                GenomicRanges_1.36.0       </span></a>
<a class="sourceLine" id="cb17-37" data-line-number="37"><span class="co">#&gt; [43] rprojroot_1.3-2             grid_3.6.0                  cli_1.1.0                  </span></a>
<a class="sourceLine" id="cb17-38" data-line-number="38"><span class="co">#&gt; [46] tools_3.6.0                 bitops_1.0-6                magrittr_1.5               </span></a>
<a class="sourceLine" id="cb17-39" data-line-number="39"><span class="co">#&gt; [49] RCurl_1.95-4.12             tibble_2.1.1                crayon_1.3.4               </span></a>
<a class="sourceLine" id="cb17-40" data-line-number="40"><span class="co">#&gt; [52] pkgconfig_2.0.2             MASS_7.3-51.1               pheatmap_1.0.12            </span></a>
<a class="sourceLine" id="cb17-41" data-line-number="41"><span class="co">#&gt; [55] Matrix_1.2-17               xml2_1.2.0                  assertthat_0.2.1           </span></a>
<a class="sourceLine" id="cb17-42" data-line-number="42"><span class="co">#&gt; [58] rmarkdown_1.12              roxygen2_6.1.1              extraDistr_1.8.10          </span></a>
<a class="sourceLine" id="cb17-43" data-line-number="43"><span class="co">#&gt; [61] rstudioapi_0.10             R6_2.4.0                    compiler_3.6.0</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#proda">proDA</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#quickstart">Quickstart</a></li>
      <li><a href="#proda-walkthrough">proDA Walkthrough</a></li>
      </ul>
</li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Constantin Ahlmann-Eltze.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
